src/script/api/script_group.hpp

```diff
@@ -128,7 +128,9 @@ class ScriptGroup : public ScriptObject {
 	 * Get the number of engines in a given group.
 	 * @param group_id The group to get the number of engines in.
 	 * @param engine_id The engine id to count.
+	 * @pre ScriptEngine::IsValidEngine(engine_id).
 	 * @pre IsValidGroup(group_id) || group_id == GROUP_ALL || group_id == GROUP_DEFAULT.
+	 * @pre !IsValidGroup(group_id) || ScriptEngine::GetVehicleType(engine_id) == GetVehicleType(group_id).
```

Comment:
`!IsValidGroup(group_id)` conflicts with the line above. Isn't `valid_group` checked in the code just to short_circuit GetVehicleType() if the group is invalid?

I think the precondition is just ensuring the type is the same.
```suggestion
	 * @pre ScriptEngine::GetVehicleType(engine_id) == GetVehicleType(group_id).
```

Reply:
`GROUP_ALL` and `GROUP_DEFAULT` are not valid groups. For these two, the vehicle type is extracted inside `GetGroupNumEngines`, from the `engine_id`.

Reply:
I'm not questioning your code, but the precondition documentation (which does not match the code).

Reply:
Sorry, I disagree.
It's either
`* @pre !IsValidGroup(group_id) || ScriptEngine::GetVehicleType(engine_id) == GetVehicleType(group_id).`
or
`* @pre group_id == GROUP_ALL || group_id == GROUP_DEFAULT || ScriptEngine::GetVehicleType(engine_id) == GetVehicleType(group_id).`

If I omit the first part, `ScriptEngine::GetVehicleType(engine_id) == GetVehicleType(group_id)` will run erroneously for `GROUP_DEFAULT` or `GROUP_ALL`.